runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/spring/ServiceBeanFactoryPostProcessor.java

```diff
@@ -0,0 +1,364 @@
+    private BindingConverterFactory        bindingConverterFactory;
+    private ConfigurableEnvironment        environment;
+
+    public ServiceBeanFactoryPostProcessor(ConfigurableApplicationContext applicationContext,
+                                           SofaRuntimeContext sofaRuntimeContext,
+                                           BindingConverterFactory bindingConverterFactory) {
+        this.applicationContext = applicationContext;
+        this.sofaRuntimeContext = sofaRuntimeContext;
+        this.bindingConverterFactory = bindingConverterFactory;
+        this.environment = applicationContext.getEnvironment();
+    }
+
+    @Override
+    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
+        Arrays.stream(beanFactory.getBeanDefinitionNames())
+            .collect(Collectors.toMap(Function.identity(), beanFactory::getBeanDefinition))
+            .forEach((key, value) -> transformSofaBeanDefinition(key, value, beanFactory));
+    }
+
+    /**
+     * {@link ScannedGenericBeanDefinition}
+     * {@link AnnotatedGenericBeanDefinition}
+     * {@link GenericBeanDefinition}
+     * {@link org.springframework.beans.factory.support.ChildBeanDefinition}
+     * {@link org.springframework.beans.factory.support.RootBeanDefinition}
+     * {@link org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.ConfigurationClassBeanDefinition}
+     */
+    private void transformSofaBeanDefinition(String beanId, BeanDefinition beanDefinition,
+                                             ConfigurableListableBeanFactory beanFactory) {
+        if (isFromConfigurationSource(beanDefinition)) {
+            generateSofaServiceDefinitionOnMethod(beanId, (AnnotatedBeanDefinition) beanDefinition,
+                beanFactory);
+        } else {
+            Class<?> beanClassType = resolveBeanClassType(beanDefinition);
+            if (beanClassType == null) {
+                SofaLogger.warn("Bean class type cant be resolved from bean of {}", beanId);
+                return;
+            }
+            generateSofaServiceDefinitionOnClass(beanId, beanClassType, beanDefinition, beanFactory);
+        }
+    }
+
+    private void generateSofaServiceDefinitionOnMethod(String beanId,
+                                                       AnnotatedBeanDefinition beanDefinition,
+                                                       ConfigurableListableBeanFactory beanFactory) {
+        Class<?> returnType;
+        Class<?> declaringClass;
+        Method method = null;
+
+        MethodMetadata methodMetadata = beanDefinition.getFactoryMethodMetadata();
+        try {
+            returnType = ClassUtils.forName(methodMetadata.getReturnTypeName(), null);
+            declaringClass = ClassUtils.forName(methodMetadata.getDeclaringClassName(), null);
+            for (Method m : declaringClass.getDeclaredMethods()) {
+                if (!m.getName().equals(beanId)) {
```

Comment:
when use `@Bean("customBeanName")`, beadId not equals methodName.

Reply:
fix.